app/react/src/server/cra_config.js

```diff
@@ -0,0 +1,83 @@
+import semver from 'semver';
+
+let MiniCssExtractPlugin;
+let normalizeCondition;
+
+export function isReactScriptsInstalled() {
+  try {
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies
+    const reactScriptsJson = require('react-scripts/package.json');
+    if (semver.lt(reactScriptsJson.version, '2.0.0')) return false;
+
+    // eslint-disable-next-line global-require, import/no-extraneous-dependencies, prefer-destructuring
+    normalizeCondition = require('webpack/lib/RuleSet').normalizeCondition;
```

Comment:
Why do you need to save `normalizeCondition` as a module state? Can't we just import it?

Reply:
We need `normalizeCondition` in both `getStyleRules` and `applyCRAWebpackConfig` but I only wanted to import it once. My rationale was that if we import at the beginning of the file, and `react-scripts` **isn't** a dependency, then we could get errors.

Though now that I think about it, this is more necessary for the `mini-css-extract-plugin` below. Webpack **shouldn't** ever error since it's a dependency of `@storybook/core`. However, if we import it directly, I think `webpack` should be added as a dependency of `@storybook/react`.

What are you thoughts?

Reply:
I don't see problems with adding it as a dep